poseidon2/merkle.nim

```diff
@@ -0,0 +1,45 @@
+import std/sequtils
+import constantine/math/arithmetic
+import constantine/math/io/io_fields
+import ./types
+import ./io
+import ./compress
+
+const KeyNone = F.fromHex("0x0")
+const KeyBottomLayer = F.fromHex("0x1")
+const KeyOdd = F.fromHex("0x2")
+const KeyOddAndBottomLayer = F.fromhex("0x3")
+
+func merkleRoot(xs: openArray[F], isBottomLayer: static bool) : F =
+  let a = low(xs)
```

Comment:
`merkleRoot` appears to get into an infinite loop when the input is empty. Suggestion: handle that explicitly and throw an exception that the empty input is invalid

Reply:
Good catch, thanks! Fixed in 7d02d4f